exes/mqtt-broadcaster/tests/mqtt_broadcaster_tests.cpp

```diff
@@ -0,0 +1,294 @@
+    packet_id_++;
+    return packet_id_;
+  }
+
+  // This function should be called to close the socket of the MQTT client but for mocking purposes it is not strictly needed
+  static auto close(boost::asio::use_awaitable_t<>) -> boost::asio::awaitable<void> { co_return; }
+
+  // This function is called when a program wants to wait for the next packet that is received on the MQTT client
+  auto recv(boost::asio::use_awaitable_t<>) -> boost::asio::awaitable<async_mqtt::packet_variant> {
+    if (online_) {
+      co_return async_mqtt::v5::connack_packet{ true, async_mqtt::connect_reason_code::success };
+    } else {
+      co_return async_mqtt::v5::connack_packet{ false, async_mqtt::connect_reason_code::server_unavailable };
+    }
+  }
+
+  // This function is the same as the previous one, but it allows the user to specify which packets to receive.
+  auto recv(async_mqtt::filter, std::set<async_mqtt::control_packet_type>, boost::asio::use_awaitable_t<>)
+      -> boost::asio::awaitable<async_mqtt::packet_variant> {
+    if (online_) {
+      co_return async_mqtt::v5::connack_packet{ false, async_mqtt::connect_reason_code::success };
+    } else {
+      co_return async_mqtt::v5::connack_packet{ false, async_mqtt::connect_reason_code::server_unavailable };
+    }
+  }
+
+  // This function returns the socket of the MQTT client
+  auto next_layer() -> boost::asio::ip::tcp::socket& { return socket_; }
+
+  // This function allows the testing program to mock that the MQTT client is offline
+  auto set_online(bool online) -> void { this->online_ = online; }
+
+private:
+  boost::asio::strand<boost::asio::io_context::executor_type> strand_;
+  boost::asio::ip::tcp::socket socket_;
+  boost::asio::io_context::executor_type ctx_;
+  bool online_ = true;
+  std::vector<async_mqtt::v5::publish_packet> messages_;
+  uint16_t packet_id_ = 0;
+};
+
+// This function tests the sending of a simple value (bool) to the MQTT client
+static auto send_simple_value(std::string mqtt_host,
+                              std::string mqtt_port,
+                              std::string mqtt_username,
+                              std::string mqtt_password) -> void {
+  boost::asio::io_context ctx{};
+
+  tfc::ipc_ruler::ipc_manager_client_mock ipc_client;
+
+  tfc::ipc::signal<tfc::ipc::details::type_bool, tfc::ipc_ruler::ipc_manager_client_mock> sig(ctx, ipc_client, "test_signal",
+                                                                                              "description");
+
+  const std::shared_ptr<mock_mqtt_client> mqtt_client =
+      std::make_shared<mock_mqtt_client>(async_mqtt::protocol_version::v5, ctx.get_executor());
+
+  const mqtt_broadcaster<tfc::ipc_ruler::ipc_manager_client_mock&, mock_mqtt_client> application(
+      ctx, std::move(mqtt_host), std::move(mqtt_port), std::move(mqtt_username), std::move(mqtt_password), ipc_client,
+      mqtt_client);
+
+  ctx.run_for(std::chrono::milliseconds(TIMEOUT_IN_MS));
```

Comment:
it is pretty long to wait 100 ms often in each test case. how could we make the wait shorter?

Reply:
lowering it to 25 ms